include/ck/tensor_operation/gpu/device/impl/device_grouped_gemm_xdl_splitk_cshuffle.hpp

```diff
@@ -364,22 +362,25 @@ struct DeviceGroupedGemmXdlSplitKCShuffle : public DeviceGroupedGemmSplitK<ALayo
         index_t group_count_;
         index_t skipped_group_count_;
 
-        std::vector<GemmTransKernelArg> gemm_kernel_args_;
+        GemmTransKernelArg* gemm_kernel_host_args_;
         index_t grid_size_;
     };
 
     // Invoker
     struct Invoker : public BaseInvoker
     {
-        float Run(const Argument& arg, const StreamConfig& stream_config = StreamConfig{})
+        float Run(const Argument& arg,
+                  const StreamConfig& stream_config = StreamConfig{},
+                  hipStream_t cpy_stream            = nullptr,
+                  hipEvent_t cpy_event              = nullptr)
         {
-            index_t K0                       = arg.gemm_kernel_args_[0].karg_.K0Padded;
-            bool all_have_kbatch_gt_one      = arg.gemm_kernel_args_[0].karg_.k_batch > 1;
+            index_t K0                       = arg.gemm_kernel_host_args_[0].karg_.K0Padded;
+            bool all_have_kbatch_gt_one      = arg.gemm_kernel_host_args_[0].karg_.k_batch > 1;
```

Comment:
What if the user did not pass cpy event & stream? Then you have unallocated `gemm_kernel_host_args_`